uaclient/security.py

```diff
@@ -562,6 +563,40 @@ def fix_security_issue_id(
     }
 
     if "CVE" in issue_id:
+        # Check livepatch status for CVE in fixes before checking CVE api
+        try:
+            status_stdout, _ = system.subp(
+                [
+                    "canonical-livepatch",
+                    "status",
+                    "--verbose",
+                    "--format=json",
+                ]
+            )
+        except exceptions.ProcessExecutionError:
+            pass
+        if status_stdout:
+            try:
+                parsed_patch = json.loads(status_stdout)["Status"][0][
+                    "Livepatch"
+                ]
+
+                if parsed_patch:
+                    fixes = parsed_patch.get("Fixes", [])
+                    if any(
+                        fix["Name"] is issue_id.lower() and fix["Patched"]
```

Comment:
Hello, @dixonmyles 

I think I saw something related to your test failure:
This comparison will never work, because `is` will go for the obj reference, and it will never be the same. So, you'd need to change it to:

```suggestion
                        fix["Name"] == issue_id.lower() and fix["Patched"]
```

Because this check fails, it will not have the print+return behavior expected in the block, and will continue processing. 

Then, on lines 601-602 there is a network leak - the code actually goes for the CVE/USN provided, and since you provided a real one as an example, it will return proper information about USNs and packages related.
The USNs related to this CVE won't have released packages on Jammy (my system, for instance), but on Bionic they do, and that's why you can't see the error locally but can on the CI: Our CI is actually running the test on a Bionic system.

Applying the aforementioned change fixes the test, as it returns early as it should with the given mocked information.

Reply:
@lucasmoura @orndorffgrant there you have the LSN without a source link that Grant mentioned yesterday :D
Besides that, I wonder what do we lose for not being able to run the Xenial tests on Xenial... (And glad we changed CI to cover that, if running all from same base we'd never see this - aha)

Reply:
D’oh! Grant actually mentioned the same thing about is vs == and I completely spaced it! Given that the test cases all pass locally for me, does that mean 22.04 compares it fine but not 18.04? Or was it just coincidence that the object references were the same locally?

Either way, thanks for the nudge! Two pairs of eyes are better than one, and this has been a great exercise! I’ll get that fix to the test case committed.

Reply:
Nice! I think what happened is that the `is` check did fail on 22.04 so it didn't return early and the rest of the function executed. But because `cve-2013-1798` doesn't affect the kernel on 22.04 the function still eventually returned `FixStatus.SYSTEM_NOT_VULNERABLE`. So the test passed, but not because of the code added in this PR :upside_down_face:

Reply:
The joys of software 😂